### ISSUE TITLE
Have the option to add an image encoded as base64

### ISSUE THREAD

**Issue by rl-king** [zotonic/zotonic#1759]
**Use case**
As a placeholder for a larger version of the image to be displayed. You could use a very small, like 8x10 px, base64 version as background-image in the same place the larger version. Rendering this in the template will save the extra request and will instantly add the visual element to the page. 

**Examples**
Medium.com does this by default in articles.

![base64example](https://user-images.githubusercontent.com/18724157/28180421-58e13ab2-6805-11e7-9f95-66f5b019ee3a.jpg)

Another example is this Polymer custom element called 'iron-image' that lets you define a placeholder as an attribute. This can be used as an `<img>` replacement and fades in the larger image once loaded. This, of course, is handled by Js and Css defined in the source of the element. 

```
<iron-image placeholder="data:image/gif;base64,..." src="large-image.jpeg"></iron-image>
```

**Comment by mworrell** [comment]
@rl-king and I looked into this a bit further.

It is possible to get this effect using a background image and a blur.
The background image can be very small.

Example html:

```html
<html>
    <head>
        <title>Test Pagina</title>
        <style type="text/css">
            body {
                margin: 0;
                padding: 0;
                background-color: black;
            }
            img {
                border: 0;
                width: 50vw;
                height: 33vw;
            }
            .test {
                display: inline-block;
                position: relative;
                background-position: -100% -100%;
                background-repeat: no-repeat;
            }
            .test::before {
                content: "";
                position: absolute;
                z-index: -1;
                display: block;
                filter: blur(20px);
                top: 0;
                left: 0;
                width: 100%;
                height: 100%;
                border: 0;
                background-image: inherit;
                background-position: center center;
                background-size: cover;
                overflow: hidden;
            }
        </style>
    </head>
    <body>
        <div class="test" style="background-image: url(/lib/images/test/t20t.jpg)">
            <img src="/lib/images/test/DSC01179.jpg" width="6000" height="4000">
        </div>
        <div class="test" style="background-image: url(/lib/images/test/t20t.jpg)">
            <div style="width:50vw;height:33vw;"></div>
        </div>
    </body>
</html>
```

And the resize is then like:

```
convert DSC01179.jpg -thumbnail 20x20 -quality 10 t20t.jpg
```

Which gives a JPEG of about ~320 bytes.

This is small enough to inline using a `data:` url.

If we would add an `{% image_data_url .... %}` tag then this can be done without additional network traffic.  Of course, generating the data url inline takes more effort and time on the server, which slows template rendering.

**Comment by mworrell** [comment]
Example output of the above:

![Screenshot 2020-04-02 at 12 11 11](https://user-images.githubusercontent.com/38268/78246525-96893d80-74e9-11ea-877e-d477cdae90cc.png)

**Comment by mworrell** [comment]
Closed with #2383
